src/components/Tree/Tree.stories.tsx

```diff
@@ -54,84 +21,30 @@ export default {
-            let modifiedItem = { ...item };
-
-            const shouldUpdate =
-                (index === 0 && item.sort === null) || // First item without a sort value
-                (index < movedItemTargetIndex && item.sort === null) || // Previous item without a sort value
-                index === movedItemTargetIndex || // Moved item
-                (index > movedItemTargetIndex &&
-                    item.sort !== null &&
-                    previousItem &&
-                    previousItem.sort &&
-                    item.sort < previousItem.sort); // Following item with a sort prop lower than on previousItem
-
-            if (shouldUpdate) {
-                const incrementedPreviousItemSort = previousItem && previousItem.sort ? previousItem.sort + 1 : null;
-                const updatedSortValue = index === 0 ? 0 : incrementedPreviousItemSort;
+    const [nodesState, setNodes] = useState(apiNodes);
 
-                modifiedItem = { ...item, sort: updatedSortValue };
-                changedItemsIds.push(item.id);
+    const handleMove = (modifiedItems: DraggableItem<TreeListItem>[]) => {
+        const modifiedArray = nodesState.map((item) => {
+            const matchingModifiedItem = modifiedItems.find((modifiedItem) => modifiedItem.id === item.id);
+            if (matchingModifiedItem) {
+                return { ...matchingModifiedItem };
             }
 
-            modifiedSameLevelArray.push(modifiedItem);
-            return;
+            return { ...item };
         });
 
-        return nodes.map((node) => {
-            if (changedItemsIds.includes(node.id)) {
-                return modifiedSameLevelArray.filter((item) => item.id === node.id)[0] ?? node;
-            }
-
-            return node;
-        });
-    };
-
-    const reorderItem = (targetItem: TreeNodeProps, sourceItem: TreeNodeProps, position: DropZonePosition) => {
-        let nodesClone = [...apiNodes];
-        const sourceIndex = nodesClone.findIndex((node) => node.id === sourceItem.id);
-
-        nodesClone.splice(sourceIndex, 1);
-
-        let targetIndex = nodesClone.findIndex((node) => node.id === targetItem.id);
-        const targetTreeListItem = nodesClone[targetIndex];
-
-        targetIndex =
-            position === DropZonePosition.AFTER || position === DropZonePosition.WITHIN ? targetIndex + 1 : targetIndex;
-        const parentId = position === DropZonePosition.WITHIN ? targetItem.id : targetTreeListItem.parentId;
-
-        const movedItem = { ...sourceItem, parentId, sort: null };
-        nodesClone = updateItemsPosition(nodesClone, parentId, targetIndex, movedItem);
-
-        setNodes(listToTree(nodesClone));
-        apiNodes = nodesClone;
+        setNodes(modifiedArray);
     };
 
     return (
         <div style={{ maxWidth: "800px" }}>
-            <TreeComponent {...args} nodes={nodesState} onDrop={reorderItem} />
+            <TreeComponent {...args} nodes={nodesState} handleUpdate={handleMove} />
```

Comment:
rename handleUpdate prop to onUpdate - props should have **on** prefix but actual function that handles that prop should have **handle** prefix

Reply:
Done